### PR TITLE
[DUOS-2968][risk=no] ERA Commons Component Refactor

### PR DESCRIPTION
**Pull request by rushtong**
### Addresses
https://broadworkbench.atlassian.net/browse/DUOS-2968

### Summary
* Convert to a functional component from a class based component
* Refactor the external call to the profile server to parse the token and instead, parse it ourselves
* Note that github (not git) cannot distinguish the changed file from `eRACommons.js` -> `ERACommons.jsx`.

### Testing
This will function normally from:
* Account linking on the user profile page
* Account linking on the data access request page

### Notes
The "Full DAR" tab on a voting page also uses this component (in `read-only` view) but has an existing bug so that a user with a commons id is not displayed. This is proving challenging to fix currently, so will file a new ticket that can be addressed separately from this update.

----
Have you read [Terra's Contributing Guide](https://github.com/DataBiosphere/terra-ui/wiki/Contributor-Guide) lately? If not, do that first.

- Label PR with a Jira ticket number and include a link to the ticket
- Label PR with a security risk modifier [no, low, medium, high]
- PR describes scope of changes
- Get a minimum of one thumbs worth of review, preferably two if enough team members are available
- Get PO sign-off for all non-trivial UI or workflow changes
- Verify all tests go green
- Test this change deployed correctly and works on dev environment after deployment

### CODE REVIEW

**Review by aarohinadkarni — APPROVED** [review]
looks good! 👍

File: cypress/component/ERACommons/era_commons.spec.js [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+describe('ERA Commons Utility', function () {
+  it('decodeNihToken works with a valid base64 encoded token', async function () {
+    const token = {'nih-username-token': encodedToken};
+    const decoded = await decodeNihToken(token);
+    expect(decoded).to.not.equal(null);
+    expect(decoded.eraCommonsUsername).to.equal('testing');
+    expect(decoded.iat).to.equal(1709561276);
+    expect(decoded.exp).to.equal(1709564876);
+  });
+  it ('decodeNihToken returns null if the token is invalid', async function () {
```
Comment (fboulnois) [inline cypress/component/ERACommons/era_commons.spec.js]:
```suggestion
  it('decodeNihToken returns null if the token is invalid', async function () {
```

nit: remove extra space

File: src/components/ERACommons.jsx [hunk 2]
```diff
@@ -0,0 +1,187 @@
...
+      if (props.readOnly && props.researcherProfile) {
+        setResearcherProfile(props.researcherProfile);
+        const propsData = researcherProfile.properties;
+        setCommonResearcherPropertyState(propsData, researcherProfile.eraCommonsId);
+      } else {
+        await getUserInfo();
+      }
+    };
+    fetchData();
+  }, []);
```
Comment (fboulnois) [inline src/components/ERACommons.jsx]: Is it necessary to have a separate `useEffect` for this functionality?

File: src/components/ERACommons.jsx [hunk 3]
```diff
@@ -0,0 +1,187 @@
...
+    setCommonResearcherPropertyState(updatedUserProps, decodedToken.eraCommonsUsername);
+    document.getElementById('era-commons-id').scrollIntoView(
+      {block: 'start', inline: 'nearest', behavior: 'smooth'}
+    );
+  };
+
+  const redirectToNihLogin = async () => {
+    const destination = window.location.origin + '/' + props.destination + '?nih-username-token=<token>';
+    const nihUrl = `${ await Config.getNihUrl() }?${qs.stringify({ 'return-url': destination })}`;
+    window.location.href = nihUrl;
```
Comment (fboulnois) [inline src/components/ERACommons.jsx:98]:
```suggestion
    props.history.push(nihUrl);
```

Should we use the React mechanism for redirecting instead?

**Review by fboulnois — APPROVED** [review]
A few questions, but otherwise looks good:

File: src/components/ERACommons.jsx (hunk 3 above)
Comment (rushtong) [reply src/components/ERACommons.jsx:98]: This only redirects internally, it doesn't redirect externally ... i.e., it tacks on the `nihUrl` to the base url which ends up failing. I don't see another mechanism for doing this, but would love to see something other than the `window.location.href` approach

File: src/components/ERACommons.jsx (hunk 2 above)
Comment (rushtong) [reply src/components/ERACommons.jsx]: Great point, I had a ton of `useEffect` problems when testing (infinite reloads, etc), but it turns out we can combine them.

**Review by fboulnois — APPROVED** [review]
looks good 👍

**Pull request merged**